src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,414 @@
+pragma solidity ^0.8.12;
+
+import {ECDSAStakeRegistryStorage, Quorum, StrategyParams} from "./ECDSAStakeRegistryStorage.sol";
+import {IStrategy} from "eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+import {IDelegationManager} from "eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {IServiceManager} from "../interfaces/IServiceManager.sol";
+
+import {OwnableUpgradeable} from "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import {CheckpointsUpgradeable} from "@openzeppelin-upgrades/contracts/utils/CheckpointsUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin-upgrades/contracts/security/PausableUpgradeable.sol";
+import {SignatureCheckerUpgradeable} from "@openzeppelin-upgrades/contracts/utils/cryptography/SignatureCheckerUpgradeable.sol";
+import {IERC1271Upgradeable} from "@openzeppelin-upgrades/contracts/interfaces/IERC1271Upgradeable.sol";
+
+/// @title ECDSA Stake Registry
+/// @notice THIS CONTRACT IS NOT AUDITED
+/// @notice Manages operator registration and quorum updates for an AVS using ECDSA signatures.
+/// @dev Extends OpenZeppelin's upgradeable Ownable and Pausable patterns to ensure upgradability and pause functionality.
+contract ECDSAStakeRegistry is
+    IERC1271Upgradeable,
+    OwnableUpgradeable,
+    PausableUpgradeable,
+    ECDSAStakeRegistryStorage
+{
+    using SignatureCheckerUpgradeable for address;
+    using CheckpointsUpgradeable for CheckpointsUpgradeable.History;
+
+    /// @dev Constructor to create ECDSAStakeRegistry.
+    /// @param _delegationManager Address of the DelegationManager contract that this registry interacts with.
+    constructor(
+        IDelegationManager _delegationManager
+    ) ECDSAStakeRegistryStorage(_delegationManager) {
+        // _disableInitializers();
+    }
+
+    /// @notice Initializes the contract with the given parameters.
+    /// @param _serviceManager The address of the service manager.
+    /// @param _thresholdWeightBps The threshold weight in basis points.
+    /// @param _quorum The quorum struct containing the details of the quorum thresholds.
+    function initialize(
+        address _serviceManager,
+        uint256 _thresholdWeightBps,
+        Quorum memory _quorum
+    ) external initializer {
+        __ECDSAStakeRegistry_init(_serviceManager, _thresholdWeightBps, _quorum);
+    }
+
+    /// @notice Registers a new operator using a provided signature
+    /// @param _operatorSignature Contains the operator's signature, salt, and expiry
+    function registerOperatorWithSignature(
+        address _operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory _operatorSignature
+    ) external {
+        _registerOperatorWithSig(_operator, _operatorSignature);
+    }
+
+    /// @notice Deregisters an existing operator
+    function deregisterOperator() external {
```

Comment:
I'm curious if there's any cases where we would not want an operator to be able to deregister themselves from an AVS

Reply:
yeah, I could see there being restrictions on when they could deregister themselves i.e, if they're still under some bonding commitment. but the scenarios would probably be highly specific to the AVS